infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -966,26 +965,25 @@ private LogEntry getLogEntry(long address, LogData entry) {
 
         ByteBuffer allRecordsBuf = ByteBuffer.allocate(totalBytes);
```

Comment:
One question (I know it's not related to your PR though). Will `ByteBuffer.allocateDirect()` be more efficient here?

Reply:
Yes, you are right this will eliminate a copy, but also note this issue exists https://bugs.openjdk.java.net/browse/JDK-4469299

Reply:
This whole class is a mess, I am in the process of cleaning it up, will use a pooled buffer allcoator in subsequent patches.